src/components/Header/Header.component.jsx

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import Styled from './styled';
+import MenuButton from '../MenuButton';
+import Search from '../Search';
+import ThemeToggle from '../ThemeToggle';
+import LoginAvatar from '../LoginAvatar';
+
+const Header = () => (
+  <Styled.Header>
+    <ul>
+      <Styled.LeftItem>
+        <MenuButton />
```

Comment:
Since the header is the only one that knows the organization of your components Menu, Search, Login and Toggle, and it would be the only place to use them, you could define a prop at each component to define whether is going to float left or right.

Hopefully I explained well myself, let me know if not :P